### PR TITLE
Created RadioButton component

### LINKED ISSUE #2703: (SP: 3) Create RadioButton component aligned with design system
#### Description:
Implement a reusable `RadioButton` component that meets the design system standards. This component will be used throughout the application to maintain consistency in radio button styles, behavior, and accessibility.

[Figma](https://www.figma.com/design/liJZDYhUnDP15Pi9bipDcv/Space2Study-(Students)?node-id=20967-98389&t=l5PI8JRXpNFKQBr0-4)

#### Requirements:
- **Styles:**
  - Follow design system variables for colors, typography, padding, and other styles.
  - Ensure hover, focus, active, and disabled states are defined according to the design system.
  - Provide a label for the radio button
  
- **Props:**
  - `label`: Provide a label for the radio button.
  - `labelPosition`: Supports different positions (bottom, top, end)
  -  `color`: Supports different colors (primary, secondary, error, success)
  - `size`: Supports different sizes (sm, md, lg).
  - `loading`: Disables the radio button when true. When loading, the  radio button shouldn't shrink or change its size compared to its default state
  - `disabled`: Disables the radio button when true.

Don't forget to cover component with tests and add to the storybook

### PR DESCRIPTION
**Pull request by SofiiaYevush**
Created RadioButton component that meets the design system standards. #2703 

**Examples of RadioButton in Storybook:**

1. image 
3. image 
3. image
[ img1, img2, img3: Relevant size, color, label position, checked:true ]

4. image
[ img4: checked:false ]

5. image
[ img5: disabled:true ]

6. image
[ img6: loading:true ]

### CODE REVIEW

**Review by ShadowOfTheSpace — APPROVED** [review]

**Review by Mav-Ivan — CHANGES_REQUESTED** [review]
move component to src/design-system/components
move stories to src/design-system/stories
move test to tests/unit/design-system
Also use aliases `~scss-components `, `~scss`

**Comment by SofiiaYevush** [comment]
> move component to src/design-system/components move stories to src/design-system/stories move test to tests/unit/design-system Also use aliases `~scss-components `, `~scss`

Done. 
Moved: `RadioButton.tsx, RadioButton.scss` to `src/design-system/components/radio-button`;
`RadioButton.stories.tsx` to `src/design-system/stories`;
`RadioButton.spec.jsx` to `tests/unit/design-system/components/radio-button`.
Added `~scss-components `, `~scss`.

File: src/design-system/components/radio-button/RadioButton.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      disabled = false,
+      loading = false,
+      checked = false,
+      ...props
+    },
+    forwardedRef
+  ) => {
+    const isDisabled = disabled || loading
+
+    const loader = loading ? (
```
Comment (Mav-Ivan) [inline src/design-system/components/radio-button/RadioButton.tsx]: add loading && (...

**Review by Mav-Ivan — APPROVED** [review]

File: src/design-system/components/radio-button/RadioButton.tsx (hunk 1 above)
Comment (SofiiaYevush) [reply src/design-system/components/radio-button/RadioButton.tsx]: > add loading && (...

Done. Changed logic for using loading.

**Review by amoutens — APPROVED** [review]

**Pull request merged**